File: UnityScripts/Weather and Enviroment/SunLight.js
var sunSpeed : float = 2;
private var sunAngle : float = 0;
var fadeRange : float = 35;
var maxLightIntensity : float = .6;
var minLightIntensity : float = 0;
var maxAmbient : float = .45;
var minAmbient : float = .15;
var maxSkyboxBrightness : float = 1;
var minSkyboxBrightness : float = .1;
var sunsetColor : Color;
var dayColor : Color;
private var speedMultiply : float;

function Update () {
	speedMultiply = Input.GetKey("f")? 20 : 1;
	
	sunAngle = (sunAngle + sunSpeed*speedMultiply*Time.deltaTime) % 360;
	light.intensity = sunAngle < 90?
		SuperLerp (minLightIntensity, maxLightIntensity, 0.0, fadeRange, sunAngle)
		: SuperLerp (maxLightIntensity, minLightIntensity, 180.0-fadeRange, 180.0, sunAngle);
	light.color = sunAngle < 90?
		SuperColorLerp (sunsetColor, dayColor, 0.0, fadeRange, sunAngle)
		: SuperColorLerp (dayColor, sunsetColor, 180.0-fadeRange, 180.0, sunAngle);
	RenderSettings.ambientLight = sunAngle < 90?
		Gray (SuperLerp (minAmbient, maxAmbient, 0.0, fadeRange, sunAngle))
		: Gray (SuperLerp (maxAmbient, minAmbient, 180.0-fadeRange, 180.0, sunAngle));
	RenderSettings.skybox.SetColor("_Tint", sunAngle < 90?
		Gray (SuperLerp (minSkyboxBrightness, maxSkyboxBrightness, 0.0, fadeRange, sunAngle))
		: Gray (SuperLerp (maxSkyboxBrightness, minSkyboxBrightness, 180.0-fadeRange, 180.0, sunAngle))
	);
	light.enabled = sunAngle < 180? true : false;
	
	transform.eulerAngles = Vector3(sunAngle, 0, 0);
}

function SuperLerp (a : float, b : float, c : float, d : float, t : float) : float {
	return Mathf.Lerp(a, b, Mathf.InverseLerp(c, d, t));
}

function SuperColorLerp (a : Color, b : Color, c : float, d : float, t : float) : Color {
	return Color.Lerp(a, b, Mathf.InverseLerp(c, d, t));
}

function Gray (t : float) : Color {
	return Color(t, t, t);
}

File: UnityScripts/Weather and Enviroment/SunLight.cs
using UnityEngine;
using System.Collections;

public class SunLight : MonoBehaviour {

    private float _SunAngle = 0;
    public float SunSpeed = 2;
    public float FadeRange = 35;
    public float MaxLightIntensity = 6;
    public float MinLightIntensity = 0;
    public float MaxAmbient = 0.45f;
    public float MinAmbient = 0.15f;
    public float MaxSkyboxBrightness = 1;
    public float MinSkyboxBrightness = 0.1f;
    public Color SunsetColor;
    public Color DayColor;
    public float TimeCompression = 1;
    
	
	// Update is called once per frame
	void Update () {

        _SunAngle = (_SunAngle + SunSpeed * TimeCompression * Time.deltaTime) % 360;
        transform.RotateAround(Vector3.zero, Vector3.left, 20 * Time.deltaTime);

        light.intensity = _SunAngle < 90 ? 
            MathHelper.SuperLerp(MinLightIntensity, MaxLightIntensity, 0.0f, FadeRange, _SunAngle) :
            MathHelper.SuperLerp(MaxLightIntensity, MinLightIntensity, 180 - FadeRange, 180, _SunAngle);
        light.color = _SunAngle < 90 ?
            MathHelper.SuperLerp(SunsetColor, DayColor, 0.0f, FadeRange, _SunAngle) :
            MathHelper.SuperLerp(DayColor, SunsetColor, 180 - FadeRange, 180, _SunAngle);
        RenderSettings.ambientLight = _SunAngle < 90 ?
            MathHelper.Gray(MathHelper.SuperLerp(MinAmbient, MaxAmbient, 0.0f, FadeRange, _SunAngle)) :
            MathHelper.Gray(MathHelper.SuperLerp(MaxAmbient, MinAmbient, 180 - FadeRange, 180, _SunAngle));
        RenderSettings.skybox.SetColor("_Tint", _SunAngle < 90 ?
            MathHelper.Gray(MathHelper.SuperLerp(MinSkyboxBrightness, MaxSkyboxBrightness, 0.0f, FadeRange, _SunAngle)) :
            MathHelper.Gray(MathHelper.SuperLerp(MaxSkyboxBrightness, MinSkyboxBrightness, 180 - FadeRange, 180, _SunAngle)));
        //light.enabled = _SunAngle < 180;

        transform.eulerAngles = new Vector3(_SunAngle, 0, 0);



	}


}
